airflow/providers/amazon/aws/hooks/glacier.py

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from typing import Any, Dict
+
+from botocore.exceptions import ClientError
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class GlacierHook(AwsBaseHook):
+    """
+    Hook for connection with Amazon Glacier
+    """
+
+    def __init__(self, *, aws_conn_id: str = "aws_default", **kwargs) -> None:
+        super().__init__(client_type="glacier", **kwargs)
+        self.aws_conn_id = aws_conn_id
+
+    def retrieve_inventory(self, vault_name: str) -> Dict[str, Any]:
+        """
+        Initiate an Amazon Glacier inventory-retrieval job
+
+        :param vault_name: the Glacier vault on which job is executed
+        :type vault_name: str
+        """
+        job_parms = {'Type': 'inventory-retrieval'}
+        self.log.info("Retrieving inventory for vault: %s", vault_name)
+        try:
+            response = self.get_conn().initiate_job(vaultName=vault_name, jobParameters=job_parms)
+        except ClientError as e:
+            self.log.exception(e)
+            raise
+
+        if response is not None:
+            self.log.info("Initiated inventory-retrieval job for: %s", vault_name)
+            self.log.info("Retrieval Job ID: %s", response["jobId"])
+        return response
```

Comment:
Should we fail if the response is `None`? Is it even possible?

Reply:
TBH, I don't know. I can't find information if response can be None. 
Going deeper into this method `initiate_job()` I see that the response can be `UnexpectedHTTPResponseError`, but there is always response different than None.

Reply:
I would assume that the response will be an object or an exception will be thrown. @feluelle can you help as you are more familiar with AWS stuff?

Reply:
I agree.

Reply:
if `initiate_job` raises an exception here the local variable `response` will be `None`.

Reply:
`initiate_job` would contain the response context which you can get via `ClientError` i.e. `e`.
but if you re-`raise` this exception it won't come to this point of `if response is not None`.

Reply:
```python
        except ClientError as e:
            self.log.exception(e)
            raise
```
I don't understand what you are trying to do with this kind of _handling_ of the exception. I think it will be logged twice and then exits because of the re-raise.
Either you want to handle the exception (not re-raise) just by logging to notify the caller and let him fix it. Or you don't want to handle it at all (no `except` block).

Reply:
Thanks @feluelle for comments. Are suggesting to raise exception or log exception?

Reply:
Is there something you can handle? **Why did you put it there in the first place i.e what specific errors did you get?** Maybe you want to handle different errors differently. If you don't know how to handle it, remove the except block. If you think a specific exception should not lead to failure of the task it was executed by, `except` the error and only log it but do not re-raise.

Reply:
And sorry I don't have any experience with Glacier itself so I can't give you any tips on the error handling for glaciers common errors and how to handle (or fix) them.
But it is totally fine if you don't handle the errors at all. This can come in further PRs for Glacier. 👍 :)

Reply:
TBH, I don't know how to handle these exception, and I don't know if I should. In my opinion exception should terminate the task. If the error occurred the code will terminated, so handling exceptions is not needed, I think. 
AWS API will throw exception itself when the error occurred. They have well documented API. I will remove try/exceptions.
Thanks a lot Felix.

Reply:
SGTM :)